cpp/src/copying/concatenate.cu

```diff
@@ -330,12 +330,21 @@ std::unique_ptr<column> concatenate(std::vector<column_view> const& columns_to_c
 {
   CUDF_EXPECTS(not columns_to_concat.empty(), "Unexpected empty list of columns to concatenate.");
 
-  data_type const type = columns_to_concat.front().type();
+  auto const& type = columns_to_concat.front().type();
   CUDF_EXPECTS(std::all_of(columns_to_concat.begin(),
                            columns_to_concat.end(),
                            [&type](auto const& c) { return c.type() == type; }),
                "Type mismatch in columns to concatenate.");
 
+  if (is_fixed_point(columns_to_concat.front().type())) {
```

Comment:
I would think that this error check would be handled by the above `data_type` check. Do two fixed-point `data_type`s with different scales compare equal? Should that be the case?

Reply:
Perhaps class `cudf::data_type` needs an `operator==` that checks the `_id` and the `_fixed_point_scale` values match.

Reply:
Right, that's what I was thinking, but I wasn't sure if two `fixed_point` types with different scales should be considered different _types_.

Reply:
`cudf::data_type` already has an `operator==`, but it only compares id. We could change it to include `scale`, but that means tests like these:
```cpp
TYPED_TEST(FixedPointTestBothReps, SimpleFixedPointColumnWrapper)
{
  using decimalXX = fixed_point<TypeParam, Radix::BASE_10>;

  auto const a = cudf::test::fixed_point_column_wrapper<int32_t>{{11, 22, 33}, scale_type{-1}};
  auto const b = cudf::test::fixed_point_column_wrapper<int32_t>{{110, 220, 330}, scale_type{-2}};

  CUDF_TEST_EXPECT_COLUMNS_EQUAL(a, b);
}
```
would fail.

Reply:
I guess it's a matter of whether we consider fixed point columns of different scales the same type and therefore comparable.  Is this a case for EXPECT_COLUMNS_EQUIVALENT?  I'm not sure.

Reply:
scale is a template argument in [`class cnl::scaled_integer< Rep, Scale >`](https://johnmcfarlane.github.io/cnl/classcnl_1_1scaled__integer.html)
Inherently different scale means, different types in cnl. I expect same rule should apply here.

Reply:
> Is this a case for EXPECT_COLUMNS_EQUIVALENT?

Yes, I think so. The default should be to use "equivalent" unless there's a specific reason to use "equal".

Reply:
I have updated `operator==` of `data_type` and the tests. Means I could remove the check I added in `concatenate.cu`.